ontology_processing/graph_creation/make_graph_class.py

```diff
@@ -0,0 +1,652 @@
+            edge_attr["cyto_classes"] = []
+            if "risk solution" in self.B_annotated.nodes[start] or "risk solution" in self.B_annotated.nodes[end]:
+                edge_attr["cyto_classes"].append("solution-edge")
+            elif not data["properties"]:
+                # Edge is not connecting to risk solution! Check if it has sources.
+                edge_attr["cyto_classes"].append("edge-no-source")
+
+        # Extract x y positions using graphviz dot algo. Use x,y positions to make cytoscape graph
+        # Also doing some preprocessing
+        for node, data in self.B_annotated.nodes(data=True):
+
+            self.B_annotated.nodes[node]["cyto_classes"] = []
+
+            risk_or_personal_value_node = False
+            if "risk solution" in data:
+                self.B_annotated.nodes[node]["cyto_classes"].append("risk-solution")
+
+            if any(data["personal_values_10"]):
+                self.B_annotated.nodes[node]["cyto_classes"].append("personal-value")
+
+            if risk_or_personal_value_node:
+                if data.get("properties", {}).get("schema_longDescription", ""):
+                    self.B_annotated.nodes[node]["cyto_classes"].append("no-long-description")
+
+                for source in SOURCE_TYPES:
+                    if data["properties"][source]:
+                        self.B_annotated.nodes[node]["cyto_classes"].append("node-no-sources")
+
+    
+    def create_subgraph(self):
+        self.subgraph_upstream = custom_bfs(
+            self.B_annotated, "increase in greenhouse effect", "reverse"
+        ).copy()
+
+    def get_subgraphs_for_viz(self):
+        """
+        Get all the nodes that are downstream of 'increase in greenhouse effect'. 
+        should be all the impact/effect node... could probably get these by doing 
+        a class search too. Also includes nodes that are "has exposure dependencies of", 
+        like "person is in the marines", 'person is in a community likely without air conditioning', or
+        'person is elderly' so not exclusive to risks + adaptations.
+        This subgraph also contains cytoscape annotations (like cyto_classes), 
+        so shouldn't query the properties as they are not reflective of webprotege.
+        """
+        subgraph_downstream_adaptations = custom_bfs(
+            self.B_annotated, "increase in greenhouse effect", edge_type="any"
+        ).copy()
+
+        # The only difference between subgraph_downstream_adaptations and subgraph_downstream is that my
+        # subgraph_downstream excludes all adaptation solutions.
+        # Only "causation" edges would exclude "person is elderly" or "person is outside often"
+        subgraph_downstream = custom_bfs(
+            self.B_annotated, "increase in greenhouse effect", edge_type="causes_or_promotes"
+        ).copy()
+
+        total_adaptation_nodes = []
+        for effectNode in subgraph_downstream_adaptations.nodes():
```

Comment:
I think Henry swapped this part out to include the Subgraph object he made and removed the code I used to because it would be redundant? Check previous history of the code to see. What I did might be more intuitive? I think mixing the visualization modifications with the modifications used for the main app isn’t best. We want the both to be included in the processing pipeline but not tangled up within each other... best to separate them and have them occur sequentially.